File: Front/buenSaborFront/src/assets/entidades/ArticuloInsumo.ts
import Imagen from "./Imagen";
import Categoria from "./Categoria";
import UnidadMedida from "./UnidadMedida"

export default class ArticuloInsumo{
    id: number = 0;
    denominacion:string = "";
    precioVenta:number = 0;
    precioCompra:number = 0;
    stockActual:number = 0;
    stockMaximo:number = 0;
    esParaElaborar:boolean = false;
    imagen:Imagen[] = [];
    unidadMedida:UnidadMedida = new UnidadMedida();
    categoria:Categoria = new Categoria();
}

File: Back/src/main/java/com/example/buensaborback/domain/entities/ArticuloInsumo.java
package com.example.buensaborback.domain.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.envers.Audited;

import java.util.HashSet;
import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@ToString
@SuperBuilder
@Audited
public class ArticuloInsumo extends Articulo {

    private Double precioCompra;
    private Integer stockActual;
    private Integer stockMaximo;
    private boolean esParaElaborar;

//    @OneToMany(cascade = CascadeType.ALL, mappedBy = "articuloInsumo", fetch = FetchType.LAZY)
//    private Set<ArticuloManufacturadoDetalle> articuloManufacturadoDetalles = new HashSet<>();
}
